pyccel/parser/syntactic.py

```diff
@@ -643,6 +647,21 @@ def fill_types(ls):
             return container
 
         decorators = {}
+
+        # add the decorator @types if the arguments are annotated
+        annotated_args = []
+        for a in arguments:
+            if isinstance(a, Argument):
+                annotated_args.append(a.annotation)
+            elif isinstance(a, ValuedArgument):
+                annotated_args.append(a.argument.annotation)
+
+        if all(not a==Nil() for a in annotated_args):
```

Comment:
Is `__eq__` implemented for `Nil()`? I think we usually write `isinstance(a, Nil)`

Reply:
done